packages/dma-library/src/operations/refinance/README.md

```diff
@@ -0,0 +1,109 @@
+# Refinance Operations
+
+The refinance operations framework provides a way of adding new refinance operations easily for new
+protocols.
+
+The framework divides each operation in 5 steps:
+
+- Closing of the previous position (customisable)
+- Swapping of the collateral into the new collateral (if needed)
+- Opening of the new position (customisable)
+- Swapping of the new debt into the old debt (if needed, to repay the flashloan)
+- Returning of the funds to the user
+
+By doing this, the intention is to allow for composition of this operations among the different
+protocols. This way we can implement a refinance operation between AAVE and Ajna, for example as:
+
+- Close current position in AAVE for ETH/USDC
+- Swap the ETH collateral into DAI
+- Open a new position in Ajna for DAI/rETH
+- Swap the rETH debt into ETH to repay the flashloan
+- Return the remaining ETH to the user
+
+# How to add a new protocol
+
+The framework allows to add new protocols by implementing the `RefinancePartialOperationGenerator`
+interface for the new Operation. This interface defines a function that receives the list of
+arguments needed for the refinance operation and returns a list of calls to be executed. It also
+returns the value of the last storage index that was used in that set of calls. This is used to keep
```

Comment:
I don't fully understand that

Reply:
Mmm, I'm not sure which part of it but I'll try in a different way:

```
To add a new protocol to the framework, the implemented needs to add 2 functions to it for Closing a position and Opening a position. This 2 functions must have the `RefinancePartialOperationGenerator` type that receives the list of arguments needed for a refinance operation (which are common among all protocols) and returns the list of calls for the sub-operation (close or open) plus the last storage index used in the actions. This index is used to keep track of the last storage slot used by the actions so the next sub-operation can know where to take the values from if needed.
```

Let me know if this clarifies it. Otherwise we can take it only and I can update the text afterwards